x-pack/plugins/ml/public/application/data_frame_analytics/common/fields.ts

```diff
@@ -362,18 +362,16 @@ export const getDefaultSelectableFields = (docs: EsDoc[], resultsField: string):
   }
 
   const newDocFields = getFlattenedFields(docs[0]._source, resultsField);
-  return newDocFields
-    .filter(k => {
-      if (k === `${resultsField}.outlier_score`) {
-        return true;
-      }
-      if (k.split('.')[0] === resultsField) {
-        return false;
-      }
+  return newDocFields.filter(k => {
+    if (k === `${resultsField}.outlier_score`) {
+      return true;
+    }
+    if (k.split('.')[0] === resultsField) {
+      return false;
+    }
 
-      return docs.some(row => row._source[k] !== null);
-    })
-    .slice(0, MAX_COLUMNS);
+    return docs.some(row => row._source[k] !== null);
```

Comment:
For the removed `.slice()` here - do we not need to limit to a particular number of columns initially anymore?

Reply:
I plan to re-introduce that in a follow-up, beginning with transforms we thought it's fine, but for indices like filebeat data grid has a hard time handling the load.